zerver/views/auth.py

```diff
@@ -104,8 +104,7 @@ def maybe_send_to_registration(request: HttpRequest, email: str, full_name: str=
         if multiuse_object_key is not None:
             del request.session["multiuse_object_key"]
             request.session.modified = True
-            if streams_to_subscribe is not None:
-                prereg_user.streams.set(streams_to_subscribe)
+            prereg_user.streams.set(streams_to_subscribe)
```

Comment:
@hackerkid can you help me validate this change?  I'm skeptical that this is the right fix.

Reply:
Wasn't it already validated at https://github.com/zulip/zulip/pull/8231#issuecomment-369685894 ?

Reply:
Though again, not applying change to these 2 lines won't affect anything since it is already in the form of `prereg_user.streams.set()`.

Reply:
@timabbott Yup. It looks good to me. When "multiuse_object_key" is not None then streams_to_subscribe can be at worst only an empty list. See the refactoring in line 86. This is mostly a cleanup change by @umairwaheed.